source/DSCResources/DSC_CMUserDiscovery/DSC_CMUserDiscovery.psm1

```diff
@@ -0,0 +1,529 @@
+        Enabled              = $enabledStatus
+        ScheduleInterval     = $scheduleConvert.Interval
+        ScheduleCount        = $scheduleConvert.Count
+        EnableDeltaDiscovery = $deltaEnabled
+        DeltaDiscoveryMins   = $syncDelta
+        ADContainers         = $adContainerArray
+    }
+}
+
+<#
+    .SYNOPSIS
+        This will set the desired state.
+
+    .PARAMETER SiteCode
+        Specifies the site code for Configuration Manager site.
+
+    .PARAMETER Enabled
+        Specifies the enablement of the User Discovery method.
+
+    .PARAMETER EnableDeltaDiscovery
+        Indicates whether Configuration Manager discovers resources created or modified in AD DS
+        since the last discovery cycle. If you specify a value of $True for this parameter,
+        specify a value for the DeltaDiscoveryMins parameter.
+
+    .PARAMETER DeltaDiscoveryMins
+        Specifies the number of minutes for the delta discovery.
+
+    .PARAMETER ADContainers
+        Specifies an array of names of Active Directory containers to match to the discovery.
+
+    .PARAMETER ADContainersToInclude
+        Specifies an array of names of Active Directory containers to add to the discovery.
+
+    .PARAMETER ADContainersToExclude
+        Specifies an array of names of Active Directory containers to exclude to the discovery.
+
+    .PARAMETER ScheduleInterval
+        Specifies the time when the scheduled event recurs.
+
+    .PARAMETER ScheduleCount
+        Specifies how often the recur interval is run.
+#>
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $Enabled,
+
+        [Parameter()]
+        [Boolean]
+        $EnableDeltaDiscovery,
+
+        [Parameter()]
+        [ValidateRange(1,60)]
+        [UInt32]
+        $DeltaDiscoveryMins,
+
+        [Parameter()]
+        [String[]]
+        $ADContainers,
+
+        [Parameter()]
+        [String[]]
+        $ADContainersToInclude,
+
+        [Parameter()]
+        [String[]]
+        $ADContainersToExclude,
+
+        [Parameter()]
+        [ValidateSet('None','Days','Hours','Minutes')]
+        [String]
+        $ScheduleInterval,
+
+        [Parameter()]
+        [UInt32]
+        $ScheduleCount
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+
+    try
+    {
+        $state = Get-TargetResource -SiteCode $SiteCode -Enabled $Enabled
+
+        if ($Enabled -eq $true)
+        {
+            if (($PSBoundParameters.DeltaDiscoveryMins) -and ($PSBoundParameters.EnableDeltaDiscovery -eq $false -or
+                ($state.EnableDeltaDiscovery -eq $false -and
+                [string]::IsNullOrEmpty($PSBoundParameters.EnableDeltaDiscovery))))
+            {
+                throw $script:localizedData.MissingDeltaDiscovery
+            }
+
```

Comment:
if delta is disable and reenabled and DeltaMins should be specified.  Need to check System if you want to as well.